vita3k/kernel/src/load_self.cpp

```diff
@@ -74,30 +75,24 @@ static bool load_var_imports(const uint32_t *nids, const Ptr<uint32_t> *entries,
 
         const char *const name = import_name(nid);
         Address export_address;
-        {
-            const std::lock_guard<std::mutex> guard(kernel.export_nids_mutex);
-            const ExportNids::iterator export_address_it = kernel.export_nids.find(nid);
-            if (export_address_it != kernel.export_nids.end()) {
-                export_address = export_address_it->second;
-                if (kernel.late_binding_infos.contains(nid)) {
-                    kernel.late_binding_infos.emplace(nid, late_binding_info({ var_reloc_entries, reloc_size, module_id }));
-                    LOG_WARN("\tNID NOT FOUND AGAIN {} ({}) at {}, setting to stub value {}", log_hex(nid), name, log_hex(entry.address()), *Ptr<uint32_t>(export_address).get(mem));
-                }
-            } else {
-                constexpr auto STUB_SYMVAL = 0xDEADBEEF;
-                LOG_WARN("\tNID NOT FOUND {} ({}) at {}, setting to stub value {}", log_hex(nid), name, log_hex(entry.address()), log_hex(STUB_SYMVAL));
+        kernel.var_binding_infos.emplace(nid, VarBindingInfo{ var_reloc_entries, reloc_size, module_id });
+        const ExportNids::iterator export_address_it = kernel.export_nids.find(nid);
+        if (export_address_it != kernel.export_nids.end()) {
+            export_address = export_address_it->second;
+        } else {
+            constexpr auto STUB_SYMVAL = 0xDEADBEEF;
+            LOG_WARN("\tNID NOT FOUND {} ({}) at {}, setting to stub value {}", log_hex(nid), name, log_hex(entry.address()), log_hex(STUB_SYMVAL));
```

Comment:
log level can be changed to debug here. Unknown variable probably will be loaded and found later